cli/src/user.rs

```diff
@@ -95,24 +107,31 @@ impl User {
             None => return Err("Unknown group".to_string()),
         };
 
-        let mls_ciphertext = match group.core_group.borrow_mut().create_application_message(
-            &group.group_aad,
-            msg.as_bytes(),
-            &self.identity.borrow().credential,
-            PADDING_SIZE,
-            &self.crypto,
-        ) {
+        let message_out = match group
+            .mls_group
+            .borrow_mut()
+            .create_message(&self.crypto, msg.as_bytes())
+        {
             Ok(m) => m,
             Err(e) => return Err(format!("{}", e)),
         };
 
-        // Send mls_ciphertext to the group
+        let mls_ciphertext = match message_out {
```

Comment:
^^ :)

Reply:
I am not sure if I follow, `message_out` now uses `map_err`, and `mls_ciphertext` does not have a `map_err` since it is a `MlsMessageOut`. I tried to clarify the `match` by using `MlsMessageOut::Plaintext(_)` in the second arm in place of the `_`.

Reply:
You are right, my bad. The alternative would be to write it with `if let...`, but it doesn't make a big difference in this case:

```rust
let mls_ciphertext = if let MlsMessageOut::Ciphertext(ciphertext) = message_out {
    *ciphertext
} else {
    return Err("Expected MlsCiphertext, found MlsPlaintext!".into());
};
```